kong/plugins/key-auth/handler.lua

```diff
@@ -168,15 +129,45 @@ local function do_authentication(conf)
   -- retrieve the consumer linked to this API key, to set appropriate headers
   local consumer_cache_key, consumer
   consumer_cache_key = kong.db.consumers:cache_key(credential.consumer.id)
-  consumer, err      = cache:get(consumer_cache_key, nil, load_consumer,
-                                 credential.consumer.id)
+  consumer, err      = cache:get(consumer_cache_key, nil, load_consumer, credential.consumer.id)
   if err then
     kong.log.err(err)
     return nil, { status = 500, message = "An unexpected error occurred" }
   end
 
-  set_consumer(consumer, credential)
+  if conf.verify_signature then
+    -- 4. Verify sha256
+    local secret = credential.secret
+    local now = ngx.time()
+    local sha = sha256:new()
+    local ks = key .. secret
+
+
+    local verify_sig = function(time)
+      sha:reset()
+      sha:update(ks)
+      sha:update(tostring(time))
```

Comment:
This construction is vulnerable to a length extension attack (https://en.m.wikipedia.org/wiki/Length_extension_attack). This should instead use HMAC, likely in the form of `hmac(secret, key .. time)`